torch/functional.py

```diff
@@ -237,6 +237,14 @@ def einsum(equation, *operands):
            the ellipsis dimensions are at the beginning of the output.
     operands (Tensor): The operands to compute the Einstein sum of.
 
+.. note::
+
+    This function does not perform any memory or speed optimization on the given expression.
+    So doing the same overall computation with a different formula can lead to different memory
```

Comment:
This function does not optimize the given expression, so a different formula for the same computation may run faster or consume less memory. Projects like opt_einsum (https://optimized-einsum.readthedocs.io/en/stable/) can optimize the formula for you.